### ISSUE TITLE
VS Code Speech - control start of search (2)

### ISSUE THREAD

**Issue by ilearner777** [microsoft/vscode#198330]
This was previously opened on issue #198252.

##Issue
With reference to original issue submission, the solution recommended was:

- You can configure that with `accessibility.voice.speechTimeout`

However, this does not fulfill the feature I what was requesting.
The `accessibility.voice.speechTimeout` setting in Visual Studio Code controls the timeout for cancelling speech output, not speech input. 

As a result, even after adding that option, it had no effect on delaying the time from when I stopped speaking, to the time that VS Code Speech started the search.

## Request:
Can you please provide a similar option to control the timeout for speech input when using VS Code Speech?
Example: `vscodeSpeech.voice.speechInputTimeout`: \<value\>

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by bpasero** [comment]
@ilearner777 would it help if you configure the timeout to `0` to disable auto accepting? This would put you in control of triggering the search when you know you are done. 

Can you explain how a timeout would help that delays starting of recognition? Nothing should happen when you there is no audio input when recording, but maybe I am missing something.

**Comment by ilearner777** [comment]
> @ilearner777 would it help if you configure the timeout to `0` to disable auto accepting? This would put you in control of triggering the search when you know you are done.
> 
> Can you explain how a timeout would help that delays starting of recognition? Nothing should happen when you there is no audio input when recording, but maybe I am missing something.

The behaviour of listening behaviour is stated in readme file as follows: `Note: by default we will start to search after a moment when you pause talking.` 

The pause is so short that even if I pause to think for a moment, VSCode voice does search before I finish.

I tried adding recommended option: `accessibility.voice.speechTimeout` = 0,  but it had not effect on VScode voice. 

Copilot told the reason why was because the `accessibility.voice.speechTimeout` setting in Visual Studio Code controls the timeout for cancelling speech output, not speech input. And that VScode doesn't currently support this option.

It is for this reason I am requesting it.

How to reproduce:
Add `accessibility.voice.speechTimeout`  = 0
Press microphone icon
Start speaking
Pause for 1 sec

See if VScode starts search....
If it does, then setting has no effect.

**Comment by ilearner777** [comment]
I didnot find that in my version. How do I get the version you are running?

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by ilearner777** [comment]
I downloaded insiders version and works as expected!
Thanks for your help.
You close this.
